src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
```

Comment:
why sort?
u can use `Math.max` and `Mathematic`  for this logic
```suggestion
    const maxSize = Math.max(a, b, c);
    const sum = ............

    if (maxSize >= sum + maxSize) {
```